### PR TITLE
Azure: remove Azure CLI installation step

### PR DESCRIPTION
**Pull request by chewong**
With the switch to docker buildx, we no longer need to install Azure CLI for Windows image building.

/assign @feiskyer

### CODE REVIEW

File: kubetest/aksengine.go [hunk 1]
```diff
@@ -418,38 +418,9 @@ func newAKSEngine() (*aksEngineDeployer, error) {
...
 	}
 
-	if err := c.azLogin(); err != nil {
-		return nil, err
-	}
-
 	return &c, nil
 }
 
-func (c *aksEngineDeployer) azLogin() error {
```
Comment (feiskyer) [inline kubetest/aksengine.go:428]: besides windows, do we have other jobs that need `az login`?

Comment (chewong) [reply kubetest/aksengine.go:428]: No I don't think so

Comment (feiskyer) [reply kubetest/aksengine.go:428]: OK, thanks

Comment (andyzhangx) [reply kubetest/aksengine.go:428]: actually we need `az login` in all csi driver tests:
```
Error response from daemon: manifest for k8sprow.azurecr.io/azuredisk-csi:e2e-7eba3135b85a6e1c87a2dc3314d6028a3249efff not found: manifest unknown: manifest tagged by "e2e-7eba3135b85a6e1c87a2dc3314d6028a3249efff" is not found
make[2]: Entering directory '/home/prow/go/src/sigs.k8s.io/azuredisk-csi-driver'
docker buildx rm container-builder || true
no builder "container-builder" found
docker buildx create --use --name=container-builder
container-builder
az acr login --name k8sprow
make[2]: az: Command not found
make[2]: *** [Makefile:112: azuredisk-container] Error 127
make[2]: Leaving directory '/home/prow/go/src/sigs.k8s.io/azuredisk-csi-driver'
make[1]: *** [Makefile:73: e2e-bootstrap] Error 2
make[1]: Leaving directory '/home/prow/go/src/sigs.k8s.io/azuredisk-csi-driver'
Failure [9.839 seconds]
```

https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_azuredisk-csi-driver/454/pull-azuredisk-csi-driver-e2e-multi-az/1284311965778317316

Comment (aramase) [reply kubetest/aksengine.go:428]: It is required for `secrets-store-csi-driver` windows tests too. Checking to see https://github.com/kubernetes-sigs/secrets-store-csi-driver/pull/284 if we can remove it

Comment (chewong) [reply kubetest/aksengine.go:428]: we shouldn't need az acr login anymore since kubetest does a `docker login` to the ACR registry.

**Pull request merged**